src/analytics.ts

```diff
@@ -5,18 +5,26 @@ import { TCoreAttributes, TEvents } from './types'
 type Options = {
     growthbookKey?: string
     growthbookDecryptionKey?: string
-    rudderstackKey: string
+    rudderstackKey?: string
 }
 
 export function createAnalyticsInstance(options?: Options) {
     let _growthbook: Growthbook,
-        _rudderstack: RudderStack,
+        _rudderstack: RudderStack | null,
         core_data: Partial<TCoreAttributes> = {},
         tracking_config: { [key: string]: boolean } = {},
         offline_cache: { [key: string]: { event: keyof TEvents; payload: TEvents[keyof TEvents] } } = {}
 
     const initialise = ({ growthbookKey, growthbookDecryptionKey, rudderstackKey }: Options) => {
-        _rudderstack = RudderStack.getRudderStackInstance(rudderstackKey)
+        // If Rudderstack is disabled in Firebase remote config
+        if (!rudderstackKey) {
+            // If Rudderstack is already initialized, remove the instance
+            if (_rudderstack) _rudderstack = null
+        } else {
+            // If Rudderstack is enabled in Firebase remote config but its not yet initialized
+            if (!_rudderstack) _rudderstack = RudderStack.getRudderStackInstance(rudderstackKey)
```

Comment:
This covers the following cases:

|                                         | Rudderstack initialized          | Rudderstack not initialized     |
|-----------------------------------------|----------------------------------|---------------------------------|
| Rudderstack disabled in Firebase config | Set Rudderstack instance to null | Do nothing                      |
| Rudderstack enabled in Firebase config  | Do nothing                       | Initialize Rudderstack instance |